plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineSource.java

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import com.google.common.base.MoreObjects;
+
+import org.eclipse.che.api.core.model.machine.MachineSource;
+import org.eclipse.che.api.machine.server.model.impl.MachineSourceImpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.eclipse.che.plugin.docker.machine.DockerInstanceProvider.DOCKER_IMAGE_TYPE;
+
+/**
+ * Set of helper methods that identifies docker image properties
+ *
+ * @author Sergii Kabashnyuk
+ * @author Florent Benoit
+ */
+public class DockerMachineSource extends MachineSourceImpl {
```

Comment:
Looks like you can use next classes instead of implementing pretty the same in that class
https://github.com/eclipse/che/blob/1dbb9d7c3ee96fc04cdfe04a33f2ea850d3038e3/plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/parser/DockerImageIdentifier.java#L34-L34
https://github.com/eclipse/che/blob/342248f90936e31826942edfbfa601ad7036f8f5/plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/parser/DockerImageIdentifierParser.java#L27-L27